frontend/src/components/DropboxChooser/index.tsx

```diff
@@ -0,0 +1,68 @@
+import loadable from "@loadable/component";
+import React, { FC, useState } from "react";
+
+declare global {
+  interface Window {
+    Dropbox: {
+      isBrowserSupported: () => boolean;
+      choose: (options: unknown) => void;
+    };
+  }
+}
+
+const AsyncContent = loadable(
+  () =>
+    /*webpackChunkName: 'DropboxChooser/Content' */ import(
+      "src/components/DropboxChooser/components/Content"
+    )
+);
+
+type DropboxFile = {
+  link: string;
+};
+
+const DROPBOX_OPTIONS = {
+  extensions: [".h5ad"],
+  sizeLimit: 30 * 1024 * 1024 * 1024, // 30GB
+};
+
+export interface Props {
+  onSelectUploadLink: React.Dispatch<React.SetStateAction<DropboxFile["link"]>>;
+}
+
+const DropboxChooser: FC<Props> = ({ children, onSelectUploadLink }) => {
+  const [isContentShown, setIsContentShown] = useState(false);
+
+  const handleMouseOver = () => {
+    setIsContentShown(true);
+  };
+
+  const handleClick = () => {
+    if (!window.Dropbox.isBrowserSupported()) {
+      alert("Sorry, your browser does not support Dropbox upload :(");
```

Comment:
Is there an alternative option to submit if their browser is not supported?

Reply:
Great question! We don't haves alternatives right now, since Dropbox Chooser is the only UI we have. And given that our web app only supports modern browsers, they should all work for Dropbox Chooser 😄